internal/services/arckubernetes/arc_kubernetes_flux_configuration_resource_test.go

```diff
@@ -0,0 +1,710 @@
+package arckubernetes_test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kubernetesconfiguration/2022-11-01/fluxconfiguration"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ArcKubernetesFluxConfigurationResource struct{}
+
+func TestAccArcKubernetesFluxConfiguration_basic(t *testing.T) {
+	// generateKey() is a time-consuming operation, we only run this test if an env var is set.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skipf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_arc_kubernetes_flux_configuration", "test")
+	r := ArcKubernetesFluxConfigurationResource{}
+	credential := fmt.Sprintf("P@$$w0rd%d!", rand.Intn(10000))
+	privateKey, publicKey, err := ArcKubernetesClusterResource{}.generateKey()
+	if err != nil {
+		t.Fatalf("failed to generate key: %+v", err)
+	}
```

Comment:
Can we refactor this since it's duplicated in all of the other tests? We could make it a method on the `type ArcKubernetesFluxConfigurationResource struct{}` struct